assets/src/utils/use-window-width.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Hook providing the current window width as state.
+ */
+export function useWindowWidth() {
+	const [ width, setWidth ] = useState( window.innerWidth );
+
+	useEffect( () => {
+		const resizeCallback = () => {
+			setWidth( window.innerWidth );
+		};
+
+		window.addEventListener( 'resize', resizeCallback );
```

Comment:
This should be a slight optimization:

```suggestion
		window.addEventListener( 'resize', resizeCallback, { passive: true } );
```

See https://iamakulov.com/notes/resize-scroll/